agent/agent.go

```diff
@@ -550,11 +550,14 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 func (a *Agent) nodeDescriptionWithHostname(ctx context.Context, tlsInfo *api.NodeTLSInfo) (*api.NodeDescription, error) {
 	desc, err := a.config.Executor.Describe(ctx)
 
-	// Override hostname and TLS info
+	// Override hostname
 	if desc != nil {
-		if a.config.Hostname != "" && desc != nil {
+		if a.config.Hostname != "" {
 			desc.Hostname = a.config.Hostname
 		}
+		if desc.Resources != nil && a.config.GenericResources != nil {
+			desc.Resources.Generic = a.config.GenericResources
```

Comment:
I am not sure about what you mean by "collected" ?

I understand your concern about hotplug resource however, my understanding of the agent code is that this is already the current behavior (join: l.192, update: l.229).

It doesn't have a "hard mechanism" in swarmctl to add / remove (like labels) but the current behavior is the same as the resource behavior :)